.travis.yml

```diff
@@ -1,35 +1,110 @@
 sudo: required
 dist: trusty
 language: cpp
-compiler:
-- gcc
-- clang
-addons:
-  apt:
-    packages:
-    - build-essential
-    - cmake
-    - doxygen
-    - g++
-    - gcc
-    - clang
-    - graphviz
-    - libdb++-dev
-    - libdb-dev
-    - libgtest-dev
-    - libminiupnpc-dev
-    - libssl-dev
-    - libssl1.0.0
-    - libunbound-dev
-    - libunwind8-dev
-    sources:
-    - ubuntu-toolchain-r-test
-before_install:
-- sudo add-apt-repository -y ppa:kojoley/boost
-- sudo apt-get -q update
-install:
-- sudo apt-get -y install libboost-{chrono,program-options,date-time,thread,system,filesystem,regex,serialization}1.58{-dev,.0}
-script: make -j2 && HAVE_DOT=YES doxygen Doxyfile
+
+matrix:
+  include:
+
+  #
+  # Coveralls.io
+  #
+  - os: linux
+    compiler: gcc
+    addons:
+      apt:
+        packages:
+          - build-essential
+          - cmake
+          - doxygen
+          - g++
+          - gcc
+          - clang
+          - graphviz
+          - libdb++-dev
+          - libdb-dev
+          - libgtest-dev
+          - libminiupnpc-dev
+          - libssl-dev
+          - libssl1.0.0
+          - libunbound-dev
+          - libunwind8-dev
+      sources:
+        - ubuntu-toolchain-r-test
+    before_install:
+      - sudo add-apt-repository -y ppa:kojoley/boost
+      - sudo apt-get -q update
+      - pip install --user cpp-coveralls
+    install:
+      - sudo apt-get -y install libboost-{chrono,program-options,date-time,thread,system,filesystem,regex,serialization}1.58{-dev,.0}
+    script:
+      - make -j2 debug-test
+    after_success:
+      - travis_wait coveralls -e external -e tests -e cmake -e contrib -e translations -e utils --gcov-options '\-lp' &> /dev/null
+
+  #
+  # Monero release-all (gcc)
+  #
+  - os: linux
+    compiler: gcc
+    addons:
+      apt:
+        packages:
+          - build-essential
+          - cmake
+          - doxygen
+          - g++
+          - gcc
+          - clang
+          - graphviz
+          - libdb++-dev
+          - libdb-dev
+          - libgtest-dev
+          - libminiupnpc-dev
+          - libssl-dev
+          - libssl1.0.0
+          - libunbound-dev
+          - libunwind8-dev
+      sources:
+        - ubuntu-toolchain-r-test
+    before_install:
+      - sudo add-apt-repository -y ppa:kojoley/boost
+      - sudo apt-get -q update
+    install:
+      - sudo apt-get -y install libboost-{chrono,program-options,date-time,thread,system,filesystem,regex,serialization}1.58{-dev,.0}
+    script:
+      - make -j2 && HAVE_DOT=YES doxygen Doxyfile
+
+  #
+  # Monero release-all (clang)
```

Comment:
The '-' does it, doesn't it? List element of the 'matrix' list. (Just guessing).